runfolder_tests/unit/runfolder_tests.py

```diff
@@ -38,14 +40,15 @@ def test_list_available_runfolders(self):
         runfolder_svc._host = lambda: "localhost"
 
         # Test
-        runfolders = runfolder_svc.list_available_runfolders()
-        runfolders = list(runfolders)
-        self.assertEqual(len(runfolders), 2)
+        with mock.patch.object(RunfolderService, 'get_reagent_kit_barcode', return_value = None):
```

Comment:
I'm not sure that I understand why you need to mock the `get_reagent_kit_barcode` here.

Reply:
The `list_available_runfolders` will create RunFolderInfo objects and try to get the barcode from runParameters.xml (which does not exist at the dummy path). This mock will perhaps not be necessary when I replace the barcode attribute with the more general meta_data. Maybe it makes sense to set `meta_data = {}` instead of throwing an Exception if the run parameters file isn't found.

Reply:
Yeah, if we don't expect the runParameters.xml to be available then we shouldn't throw an exception. Or if we expect that it should always be there, I think that that file should be created in the dummy path if this is actually backed by files.